doozerlib/cli/release_gen_payload.py

```diff
@@ -0,0 +1,334 @@
+            error = f"Unable to find build for: {image.image_name_short}"
+            no_build_items.append(image.image_name_short)
+            state.record_image_fail(lstate, image, error, runtime.logger)
+        else:
+            for archive in archives:
+                arch = archive["arch"]
+                pullspecs = archive["extra"]["docker"]["repositories"]
+                if not pullspecs or ":" not in pullspecs[-1]:  # in case of no pullspecs or invalid format
+                    error = f"Unable to find pullspecs for: {image.image_name_short}"
+                    red_print(error, file=sys.stderr)
+                    state.record_image_fail(lstate, image, error, runtime.logger)
+                    break
+                # The tag that will be used in the imagestreams
+                tag_name = image.image_name_short
+                tag_name = tag_name[4:] if tag_name.startswith("ose-") else tag_name  # it _should_ but... to be safe
+                mirroring_value = {'version': latest_build[0]["version"], 'release': latest_build[0]["release"], 'image_src': pullspecs[-1]}
+                embargoed = latest_build[0]["id"] in embargoed_build_ids  # when public_upstreams are not configured, this is always false
+                if not embargoed:  # exclude embargoed images from the ocp[-arch] imagestreams
+                    runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the public mirroring list with imagestream tag {tag_name}...")
+                    mirroring.setdefault(arch, {})[tag_name] = mirroring_value
+                else:
+                    red_print(f"Found embargoed image {pullspecs[-1]}")
+                if runtime.group_config.public_upstreams:
+                    # when public_upstreams are configured, both embargoed and non-embargoed images should be included in the ocp[-arch]-priv imagestreams
+                    runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the private mirroring list with imagestream tag {tag_name}...")
+                    mirroring.setdefault(f"{arch}-priv", {})[tag_name] = mirroring_value
+        if not error:
+            state.record_image_success(lstate, image)
+        else:
+            red_print(error, file=sys.stderr)
+            state.record_image_fail(lstate, image, error, runtime.logger)
+
+    for key in mirroring:
+        private = key.endswith("-priv")
+        arch = key[:-5] if private else key  # strip `-priv` suffix
+
+        mirror_filename = 'src_dest.{}'.format(key)
+        imagestream_filename = 'image_stream.{}'.format(key)
+        target_is_name = is_name
+        target_is_namespace = is_namespace
+        if arch != 'x86_64':
+            target_is_name = '{}-{}'.format(target_is_name, arch)
+            target_is_namespace = '{}-{}'.format(target_is_namespace, arch)
+
+        if private:
+            target_is_namespace += "-priv"  # should we also append `-priv` suffix to target_is_name?
```

Comment:
No need to append. Imagestream names will be named with same within public & private namespaces.